qa/src/test/java/io/camunda/migrator/qa/persistence/DropSchemaTest.java

```diff
@@ -55,45 +56,68 @@ void shouldMigrationSchemaBeKeptOnShutdown() throws Exception {
   @Test
   void shouldMigrationSchemaBeDroppedOnShutdown() throws Exception {
     // given spring application is running with drop-schema flag enabled
-    var context = springApplication.properties(Map.of("camunda.migrator.drop-schema", true)).run();
+    String prefix = "FOO_";
+    var context = springApplication.properties(Map.of("camunda.migrator.table-prefix", prefix)).run("--drop-schema");
     DataSource durableDataSource = createDurableDataSource(context);
-    ensureTrue("Migration mapping table does not exist", tableExists(durableDataSource, MIGRATION_MAPPING_TABLE));
+    ensureTrue("Migration mapping table does not exist", tableExists(durableDataSource, prefix + MIGRATION_MAPPING_TABLE));
 
     // when application is shut down
     context.close();
 
     // then migration schema is dropped
-    assertThat(tableExists(durableDataSource, MIGRATION_MAPPING_TABLE)).isFalse();
+    assertThat(tableExists(durableDataSource, prefix + MIGRATION_MAPPING_TABLE)).isFalse();
   }
 
   @Test
-  void shouldMigrationSchemaBeDroppedOnShutdownWithPrefix() throws Exception {
+  void shouldMigrationSchemaBeKeptOnSkippedEntities() throws Exception {
     // given spring application is running with drop-schema flag enabled
-    var context = springApplication.properties(Map.of("camunda.migrator.drop-schema", true, "camunda.migrator.table-prefix", "FOO_")).run();
+    var context = springApplication.run("--drop-schema");
     DataSource durableDataSource = createDurableDataSource(context);
-    ensureTrue("Migration mapping table does not exist", tableExists(durableDataSource, "FOO_" + MIGRATION_MAPPING_TABLE));
+    ensureTrue("Migration mapping table does not exist", tableExists(durableDataSource, MIGRATION_MAPPING_TABLE));
+    DbClient dbClient = context.getBean("dbClient", DbClient.class);
+    // and some entities are marked as skipped
+    dbClient.insert("legacyId", null, IdKeyMapper.TYPE.RUNTIME_PROCESS_INSTANCE);
 
     // when application is shut down
     context.close();
 
-    // then migration schema is dropped
-    assertThat(tableExists(durableDataSource, "FOO_" + MIGRATION_MAPPING_TABLE)).isFalse();
+    // then migration schema is kept
+    assertThat(tableExists(durableDataSource, MIGRATION_MAPPING_TABLE)).isTrue();
+
+    // cleanup db after text
+    clearMigrationMappingTable(durableDataSource, "");
```

Comment:
🔧 If the test fails, this cleanup won't be executed. Consider moving it to `@AfterEach`.